javascript/algorithms/bubble_sort.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+var debug = require('debug')('BUBBLE SORT');
+
+var NULL = null;
+/**
+ * _isValidArray - Validate if input is array and lenght larger than 1
+ * @method _isValidArray
+ * @private
+ * @return {Boolean}  if it is valid array
+ */
+function _isValidArray(lists) {
+    return (lists && lists.constructor === Array && lists.length > 1);
+}
+
+/**
+ * Javascript implementaion of bubble sort
+ * http://en.wikipedia.org/wiki/Bubble_sort
+ * Complexity: O(n^2)
+ *
+ * @method bubbleSort
+ * @params {Array} lists target array to be sort
+ * @return {Array} list sorted array
+ */
+var bubbleSort = function bubbleSort(lists) {
+    if (!_isValidArray(lists)) {
+        debug('ERROR, INVALID ARRAY');
+        return NULL;
+    }
+
+    for (var i = 0, j = lists.length; i < j; i++) {
+        for (var a = i; a > 0; a--) {
```

Comment:
Outcome is correct. But because you are referencing to Wiki, I suppose you should implement the same algorithm. The idea is in the first run, the largest number will be determined, and then the second large, and etc.